packages/core-js/src/userop/nftMint.ts

```diff
@@ -0,0 +1,149 @@
+import GatewayClientRepository from '@/api/gateway.js';
+import type OktoClient from '@/core/index.js';
+import { BaseError } from '@/errors/base.js';
+import { getChains } from '@/explorer/chain.js';
+import type { Address, UserOp } from '@/types/core.js';
+import { Constants } from '@/utils/index.js';
+import { generateUUID, nonceToBigInt } from '@/utils/nonce.js';
+import {
+  encodeAbiParameters,
+  encodeFunctionData,
+  parseAbiParameters,
+  toHex,
+} from 'viem';
+import { INTENT_ABI } from './abi.js';
+import type { NftMintParams } from './types.js';
+import { NftMintParamsSchema, validateSchema } from './userOpInputValidator.js';
+
+/**
+ * Creates a user operation for minting an NFT.
+ *
+ * This function initiates the process of minting an NFT by encoding
+ * the necessary parameters into a User Operation. The operation is then
+ * submitted through the OktoClient for execution.
+ *
+ * @param oc - The OktoClient instance used to interact with the blockchain.
+ * @param data - The parameters for minting an NFT (caip2Id, nftName, collectionAddress, uri, and additional data).
+ * @returns The User Operation (UserOp) for the NFT minting.
+ */
+export async function nftMint(
+  oc: OktoClient,
+  data: NftMintParams,
+  feePayerAddress?: Address,
+): Promise<UserOp> {
+  if (!oc.isLoggedIn()) {
+    throw new BaseError('User not logged in');
+  }
+
+  validateSchema(NftMintParamsSchema, data);
```

Comment:
NFTMint and NftCollectionCreation is Aptos specific intents for now, please confirm with BE team and add a check for same

Reply:
aptosRawTransaction as well